lib/rules/heading-matches.js

```diff
@@ -0,0 +1,13 @@
+// Get all valid roles
+let explicitRoles;
+if (node.hasAttribute('role')) {
+  explicitRoles = node.getAttribute('role').split(/\s+/i)
+    .filter(axe.commons.aria.isValidRole);
```

Comment:
The splitting and filtering here seems like a utility we could reuse.

Reply:
Seems pretty trivial. I would like us to at some point move towards a stronger getRoles() method, but I think that needs a little more thought, and we should probably have a better idea of how we do accessibility support before we can.